utilities/esmpy/regrid_polygon_mesh.py

```diff
@@ -0,0 +1,766 @@
+import numpy as np
+import ESMF
+
+import numpy
+import netCDF4 as nc4
+import os
+import sys
+from shapely.geometry import Point, Polygon
+
+import ESMF.util.helpers as helpers
+import ESMF.api.constants as constants
+
+import glob
+from pathlib import Path
+
+from os import listdir
+from os.path import isfile, join
+import argparse
+import pandas as pd
+from  multiprocessing import Process, Lock
+import multiprocessing
+import time
+import math
+
+def get_cat_id(path):
+    """
+    Extract the id from the file path
+    """
+    path = Path(path)
+    name = path.stem
+    id = name.split('_')[0]
+    return id
+
+def mesh_create_polygon(infile, cat_id):
+    """
+    Create a ESMF mesh from the infile that contains the coordinates of the polygon
+    corresponding to a catchment
+    """
+    # Read in polygon coordinates
+    sep = " "
+    filename = infile
+    node_id = []
+    lons = []
+    lats = []
+    lons_lats = []
+    poly_coords = []
+    with open(filename) as f:
+        next(f)
+        for x in f:
+            node_id.append(int(x.split(' ')[0]))
+            lons.append(float(x.split(' ')[1]))
+            lats.append(float(x.split(' ')[2]))
+            lons_lats.append(float(x.split(' ')[1]))
```

Comment:
I assume you are talking about lons_lats. Yeah, this can be done using interleaving somewhere after the code block correcting the vertices ordering error. lons_lats can be removed from that block of code.